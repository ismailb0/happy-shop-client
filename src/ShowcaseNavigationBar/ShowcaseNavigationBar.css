.showcase-navigation {
  display: flex;
  height: 10%;
  padding-right: 3%;
  padding-top: 1%;
  padding-bottom: 1%;
  padding-left: 5%;
}
