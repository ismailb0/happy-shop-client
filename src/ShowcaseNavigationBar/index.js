export { default } from './ShowcaseNavigationBar.js';
