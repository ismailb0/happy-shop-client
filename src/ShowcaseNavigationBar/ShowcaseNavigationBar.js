import React, { Component } from 'react';

import './ShowcaseNavigationBar.css';
import Dropdown from '../Dropdown';
import Pagination from '../Pagination';

class ShowcaseNavigationBar extends Component {

SORT_CHOICES = [
  {
    label: 'Best Selling',
    value: 'best_selling',
  },
  {
    label: 'Price',
    value: 'price',
  },
]

PAGINATION_CHOICES = [
  {
    label: '10',
    value: 10,
  },
  {
    label: '30',
    value: 30,
  },
  {
    label: '50',
    value: 50,
  }
]

  getCurrentSortOption = () => {
    return this.SORT_CHOICES.find((sortOption) => sortOption.value === this.props.sortedBy)
  }

  getCurrentPaginationOption = () => {
    return this.PAGINATION_CHOICES.find((paginationOption) => paginationOption.value === this.props.resultsPerPage)
  }

  render() {
    return (
      <div className="showcase-navigation">
        <Dropdown
          inputLabel="Sort By"
          defaultItem={this.getCurrentSortOption()}
          items={this.SORT_CHOICES}
          handleChange={this.props.handleSortChange}
        />
        <Dropdown
          inputLabel="View"
          defaultItem={this.getCurrentPaginationOption()}
          items={this.PAGINATION_CHOICES}
          handleChange={this.props.handleResultsPerPageChange}
        />
        <Pagination
          handlePageChange={this.props.handlePageChange}
          totalNumberOfProducts={this.props.totalNumberOfProducts}
          resultsPerPage={this.props.resultsPerPage}
        />
      </div>
    );
  }
}

export default ShowcaseNavigationBar;
