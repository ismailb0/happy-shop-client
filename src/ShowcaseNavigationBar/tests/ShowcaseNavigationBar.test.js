import React from 'react';
import { shallow } from "enzyme";
import ShowcaseNavigationBar from '../ShowcaseNavigationBar';

describe('<ShowcaseNavigationBar />', () => {

  const props = {
    sortedBy: 'price',
    resultsPerPage: 10,
    page: 1,
    handleSortChange: jest.fn(),
    handleResultsPerPageChange: jest.fn(),
    handlePageChange: jest.fn(),
    totalNumberOfProducts: 100,
  }

  it('getCurrentSortOption should return sort option currently selected', () => {
    const renderedComponent = shallow(<ShowcaseNavigationBar {...props} />);
    const componentInstance = renderedComponent.instance()
    expect(componentInstance.getCurrentSortOption()).toEqual({
        label: 'Price',
        value: 'price',
    })
  });

  it('getCurrentPaginationOption should return pagination option currently selected', () => {
    const renderedComponent = shallow(<ShowcaseNavigationBar {...props} />);
    const componentInstance = renderedComponent.instance()
    expect(componentInstance.getCurrentPaginationOption()).toEqual({
        label: '10',
        value: 10,
    })
  });

  it('should match snapshot', () => {
    const renderedComponent = shallow(<ShowcaseNavigationBar {...props} />);
    expect(renderedComponent).toMatchSnapshot();
  });

});
