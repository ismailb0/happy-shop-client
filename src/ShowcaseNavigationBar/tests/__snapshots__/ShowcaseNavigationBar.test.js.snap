// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowcaseNavigationBar /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowcaseNavigationBar
    handlePageChange={[MockFunction]}
    handleResultsPerPageChange={[MockFunction]}
    handleSortChange={[MockFunction]}
    resultsPerPage={10}
    sortedBy="price"
    totalNumberOfProducts={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Dropdown)
          defaultItem={
            Object {
              "label": "Price",
              "value": "price",
            }
          }
          handleChange={[MockFunction]}
          inputLabel="Sort By"
          items={
            Array [
              Object {
                "label": "Best Selling",
                "value": "best_selling",
              },
              Object {
                "label": "Price",
                "value": "price",
              },
            ]
          }
        />,
        <WithStyles(Dropdown)
          defaultItem={
            Object {
              "label": "10",
              "value": 10,
            }
          }
          handleChange={[MockFunction]}
          inputLabel="View"
          items={
            Array [
              Object {
                "label": "10",
                "value": 10,
              },
              Object {
                "label": "30",
                "value": 30,
              },
              Object {
                "label": "50",
                "value": 50,
              },
            ]
          }
        />,
        <Pagination
          handlePageChange={[MockFunction]}
          resultsPerPage={10}
          totalNumberOfProducts={100}
        />,
      ],
      "className": "showcase-navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultItem": Object {
            "label": "Price",
            "value": "price",
          },
          "handleChange": [MockFunction],
          "inputLabel": "Sort By",
          "items": Array [
            Object {
              "label": "Best Selling",
              "value": "best_selling",
            },
            Object {
              "label": "Price",
              "value": "price",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "defaultItem": Object {
            "label": "10",
            "value": 10,
          },
          "handleChange": [MockFunction],
          "inputLabel": "View",
          "items": Array [
            Object {
              "label": "10",
              "value": 10,
            },
            Object {
              "label": "30",
              "value": 30,
            },
            Object {
              "label": "50",
              "value": 50,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handlePageChange": [MockFunction],
          "resultsPerPage": 10,
          "totalNumberOfProducts": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Dropdown)
            defaultItem={
              Object {
                "label": "Price",
                "value": "price",
              }
            }
            handleChange={[MockFunction]}
            inputLabel="Sort By"
            items={
              Array [
                Object {
                  "label": "Best Selling",
                  "value": "best_selling",
                },
                Object {
                  "label": "Price",
                  "value": "price",
                },
              ]
            }
          />,
          <WithStyles(Dropdown)
            defaultItem={
              Object {
                "label": "10",
                "value": 10,
              }
            }
            handleChange={[MockFunction]}
            inputLabel="View"
            items={
              Array [
                Object {
                  "label": "10",
                  "value": 10,
                },
                Object {
                  "label": "30",
                  "value": 30,
                },
                Object {
                  "label": "50",
                  "value": 50,
                },
              ]
            }
          />,
          <Pagination
            handlePageChange={[MockFunction]}
            resultsPerPage={10}
            totalNumberOfProducts={100}
          />,
        ],
        "className": "showcase-navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultItem": Object {
              "label": "Price",
              "value": "price",
            },
            "handleChange": [MockFunction],
            "inputLabel": "Sort By",
            "items": Array [
              Object {
                "label": "Best Selling",
                "value": "best_selling",
              },
              Object {
                "label": "Price",
                "value": "price",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "defaultItem": Object {
              "label": "10",
              "value": 10,
            },
            "handleChange": [MockFunction],
            "inputLabel": "View",
            "items": Array [
              Object {
                "label": "10",
                "value": 10,
              },
              Object {
                "label": "30",
                "value": 30,
              },
              Object {
                "label": "50",
                "value": 50,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handlePageChange": [MockFunction],
            "resultsPerPage": 10,
            "totalNumberOfProducts": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
